WPMedia/fusion-news-theme-blocks
# Tmedia 195 extract util trailing slash

## Description
Extract Util Functions Including Get Location and fix trailing slash

## Jira Ticket
- [TMEDIA-195](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-195&assignee=5e387d6a1b1d910e5dfd7a9b-)

## Acceptance Criteria
- Links should still work as expected without trailing slashes 

NOTE: I renamed the trailing slash method in anticipation of future ticket in this sprint changing this functionality slightly https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-4

## Test Steps

1. Checkout this branch `git checkout TMEDIA-195-extract-util`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/header-nav-block,@wpmedia/header-nav-chain-block,@wpmedia/links-bar-block,@wpmedia/overline-block,@wpmedia/shared-styles`
3. Go to Should see trailing slashes for links bar links 

![Screen Shot 2021-04-16 at 16 12 10](https://user-images.githubusercontent.com/5950956/115084495-a1cdc200-9ece-11eb-8c22-939509e8ed3e.png)
4. Should not see trailing slashes for mailto: links nor links with query params
<img width="981" alt="Screen Shot 2021-04-16 at 16 17 41" src="https://user-images.githubusercontent.com/5950956/115085019-59fb6a80-9ecf-11eb-83f2-f8a01c71656f.png">
<img width="911" alt="Screen Shot 2021-04-16 at 16 17 29" src="https://user-images.githubusercontent.com/5950956/115085020-5b2c9780-9ecf-11eb-91b8-139e6e986117.png">

## Effect Of Changes
### Before

- Link formatting duplicated and over-tested

### After

- No change, smaller bundles

## Dependencies or Side Effects

- link engine theme sdk via the `.env` file 
- Run the branch `TMEDIA-195-extract-util` in engine-theme-sdk

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
